platform/linux-generic/odp_libconfig.c

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2018, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include <stdlib.h>
+#include <libconfig.h>
+
+#include <odp_internal.h>
+#include <odp_debug_internal.h>
+#include <odp_libconfig_internal.h>
+
+extern struct odp_global_data_s odp_global_data;
+
+int _odp_libconfig_init_global(void)
+{
+	char *filename;
+	config_t *config = &odp_global_data.libconfig;
+
+	filename = getenv("ODP_CONFIG_FILE");
+	if (filename == NULL)
+		return 0;
+
+	config_init(config);
+
+	if (!config_read_file(config, filename)) {
+		ODP_ERR("Failed to read config file: %s(%d): %s\n",
+			config_error_file(config), config_error_line(config),
+			config_error_text(config));
+		config_destroy(config);
+		return -1;
+	}
+
+	odp_global_data.libconfig_enabled = 1;
+
+	return 0;
+}
+
+int _odp_libconfig_term_global(void)
+{
+	if (odp_global_data.libconfig_enabled)
+		config_destroy(&odp_global_data.libconfig);
+	return 0;
+}
+
+config_setting_t *_odp_libconfig_lookup(const char *path)
```

Comment:
This is just an internal helper to avoid having to reference to odp_global_data when calling `config_lookup()`. It could actually be removed altogether.

Since we are using libconfig only internally, what's the benefit of wrapping its functions (other than avoiding hard dependency)? We are not wrapping e.g. openssl functions.